bus/src/test/java/org.hawkular.inventory.bus/BusTest.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.inventory.bus;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.hawkular.bus.common.ConnectionContextFactory;
+import org.hawkular.bus.common.Endpoint;
+import org.hawkular.bus.common.MessageProcessor;
+import org.hawkular.bus.common.consumer.ConsumerConnectionContext;
+import org.hawkular.bus.common.producer.ProducerConnectionContext;
+import org.hawkular.bus.common.test.SimpleTestListener;
+import org.hawkular.bus.common.test.VMEmbeddedBrokerWrapper;
+import org.hawkular.inventory.api.Action;
+import org.hawkular.inventory.api.model.CanonicalPath;
+import org.hawkular.inventory.api.model.Environment;
+import org.hawkular.inventory.api.model.Feed;
+import org.hawkular.inventory.api.model.Metric;
+import org.hawkular.inventory.api.model.MetricDataType;
+import org.hawkular.inventory.api.model.MetricType;
+import org.hawkular.inventory.api.model.MetricUnit;
+import org.hawkular.inventory.api.model.Relationship;
+import org.hawkular.inventory.api.model.Resource;
+import org.hawkular.inventory.api.model.ResourceType;
+import org.hawkular.inventory.api.model.Tenant;
+import org.hawkular.inventory.bus.api.EnvironmentEvent;
+import org.hawkular.inventory.bus.api.FeedEvent;
+import org.hawkular.inventory.bus.api.MetricEvent;
+import org.hawkular.inventory.bus.api.MetricTypeEvent;
+import org.hawkular.inventory.bus.api.RelationshipEvent;
+import org.hawkular.inventory.bus.api.ResourceEvent;
+import org.hawkular.inventory.bus.api.ResourceTypeEvent;
+import org.hawkular.inventory.bus.api.TenantEvent;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author Pavol Loffay
+ * @since 0.2.0
+ */
+public class BusTest {
+
+    private static final String PROP_KEY = "p1";
+    private static final String PROP_VALUE = "prop1";
+    private static final Map<String, Object> objectProperties = new HashMap<>();
+    private static final Map<String, String> messageHeaders = new HashMap<>();
+
+    @BeforeClass
+    public static void init() {
+        objectProperties.put(PROP_KEY, PROP_VALUE);
+        messageHeaders.put(PROP_KEY, PROP_VALUE);
+    }
+
+    @Test
+    public void messagesSerializationTest() {
```

Comment:
The test currently just creates instances of the event objects. As far as I can see it doesn't do anything with them. I assume that's because of the reasons you detailed but I'd have known if there was a TODO note in the tests, too ;)

I'm just paranoid about leaving half-finished stuff lying around because there's just lots of things on our plate and things will get forgotten about if not tracked in a JIRA or at least a TODO note.